src/compiler/checker.ts

```diff
@@ -6900,12 +7083,14 @@ module ts {
             // For a tagged template, then the first argument be 'undefined' if necessary
             // because it represents a TemplateStringsArray.
             let excludeArgument: boolean[];

```

Comment:
Let's add

```TypeScript
//
// For a decorator, no argument should be susceptible to contextual typing
// because decorators are applied to *declarations*, not expressions.
```